### ISSUE TITLE
Units dropped from random() $limit

### ISSUE THREAD

**Issue by chriseppstein** [sass/sass#1890]
* [x] [Proposal](https://github.com/sass/sass/blob/main/accepted/random-with-units.md)

Deprecation

* [x] [Tests](https://github.com/sass/sass-spec/issues/1810) https://github.com/sass/sass-spec/pull/1815
* [x] [Dart Sass](https://github.com/sass/dart-sass/issues/1756) https://github.com/sass/dart-sass/pull/1779
* [x] [Docs](https://github.com/sass/sass-site/issues/659) https://github.com/sass/sass-site/pull/664
* [x] Update canonical spec https://github.com/sass/sass/pull/3377

Final

* [ ] [Tests](https://github.com/sass/sass-spec/issues/1810)
* [ ] [Dart Sass](https://github.com/sass/dart-sass/issues/1756)
* [ ] [Docs](https://github.com/sass/sass-site/issues/659)

---

_(Original:)_

The unit passed with the limit should be the unit on the returned value. Instead the unit is dropped and never considered nor is an error raised.

Step 1:
Deprecate passing units to `random()` as the `$limit`.

Step 2:
Make passing a unit to `random()` as the `$limit` an error.

Step 3:
Allow units to be passed in, returning them with the random result.

**Comment by jcutrell** [comment]
Is the desire to have `random()` be redefined to use `random($limit, $unit)`, where `$unit` is optional?

**Comment by nex3** [comment]
@jcutrell No, the desire is to (ultimately) have `math.random(10px)` return a number with the unit `px`.

**Comment by Goodwine** [comment]
Blocking until feedback arrives from proposal

https://github.com/sass/sass/blob/main/proposal/random-with-units.md

**Comment by Goodwine** [comment]
Updated main comment with:

[Proposal](https://github.com/sass/sass/blob/main/proposal/random-with-units.md)

Deprecation

* [Tests](https://github.com/sass/sass-spec/issues/1810)
* [Dart Sass](https://github.com/sass/dart-sass/issues/1756)
* [Docs](https://github.com/sass/sass-site/issues/659)
* Update canonical spec

Final

* [Tests](https://github.com/sass/sass-spec/issues/1810)
* [Dart Sass](https://github.com/sass/dart-sass/issues/1756)
* [Docs](https://github.com/sass/sass-site/issues/659)

**Comment by nex3** [comment]
Now that the proposal has landed, we'll give this a month for public comment since it's a low-controversy breaking change.

**Comment by stof** [comment]
@nex3 shouldn't this proposal be announced on the blog if you expect comments on it ?

**Comment by Goodwine** [comment]
Per the contributing guidelines
> Public comments are solicited for the feature, usually via a tweet from [@SassCSS](https://twitter.com/SassCSS). **If the feature is big enough**, a blog post soliciting feedback may also be written

Assuming that because this enhancement/fix is small and uncontroversial it doesn't cross that threshold 🤔

**Comment by Goodwine** [comment]
Deprecation added, now waiting on Dart 2.0.0 to implement spec
